pancancer_evaluation/data_models/tcga_data_model.py

```diff
@@ -509,3 +637,77 @@ def holdout_percent_labels(y,
             test_ixs |= nz_ixs
         return (train_ixs, test_ixs)
 
+    @staticmethod
+    def _get_cancers_to_add(y_df,
+                            test_cancer_type,
+                            num_cancer_types,
+                            how_to_add,
+                            similarity_matrix=None,
+                            seed=cfg.default_seed):
+
+        # start with test cancer type and add if necessary
+        train_cancer_types = [test_cancer_type]
+        # y_df should already be filtered to valid cancer types
+        valid_cancer_types = [ct for ct in y_df.DISEASE.unique()
+                                 if ct != test_cancer_type]
+
+        if num_cancer_types == -1:
+            # pan-cancer model, train on all valid cancer types
+            train_cancer_types += valid_cancer_types
+        elif num_cancer_types >= 1:
+            # add desired number of cancer types to train set
+            if how_to_add == 'random':
+                import contextlib
+                # We want this random addition of cancer types to be the same
```

Comment:
This function can be run multiple times by the script that calls it (`run_add_cancer_classification.py`). Each time I call this function, I want to choose cancers randomly in the same order (like you wrote above), so I want to reset the seed locally. But I don't want to reset the global numpy seed each time, since that might affect other things going on in the script that calls this code (e.g. cross-validation splits, model optimization, etc). Does that answer your question?